pyk/src/pyk/kast/manip.py

```diff
@@ -215,48 +215,39 @@ def extract_rhs(term: KInner) -> KInner:
 
 
 def extract_subst(term: KInner) -> tuple[Subst, KInner]:
-    def _subst_for_terms(term1: KInner, term2: KInner) -> Subst | None:
-        if type(term1) is KVariable and type(term2) not in {KToken, KVariable}:
-            return Subst({term1.name: term2})
-        if type(term2) is KVariable and type(term1) not in {KToken, KVariable}:
-            return Subst({term2.name: term1})
-        return None
-
-    def _extract_subst(conjunct: KInner) -> Subst | None:
-        if type(conjunct) is KApply:
-            if conjunct.label.name == '#Equals':
-                subst = _subst_for_terms(conjunct.args[0], conjunct.args[1])
-
-                if subst is not None:
-                    return subst
-
-                if (
-                    conjunct.args[0] == TRUE
-                    and type(conjunct.args[1]) is KApply
-                    and conjunct.args[1].label.name in {'_==K_', '_==Int_'}
-                ):
-                    subst = _subst_for_terms(conjunct.args[1].args[0], conjunct.args[1].args[1])
-
-                    if subst is not None:
-                        return subst
+    _subst = {}
+    rem_conjuncts: list[KInner] = []
 
+    def _extract_subst(_term1: KInner, _term2: KInner) -> tuple[str, KInner] | None:
+        if (
+            (type(_term1) is KVariable and _term1.name not in _subst)
```

Comment:
This is considered more Pythonic:

```suggestion
            (isinstance(_term1, KVariable) and _term1.name not in _subst)
```

([PEP8](https://github.com/python/peps/blob/2ec2140ec83dcc6069e3e757641ca82809bc87eb/peps/pep-0008.rst?plain=1#L1455-L1468))

Reply:
Taken care of in https://github.com/runtimeverification/k/pull/4630/